crates/client-api-messages/src/websocket.rs

```diff
@@ -184,6 +203,39 @@ pub struct Subscribe {
     pub request_id: u32,
 }
 
+/// Sent by client to register a subscription to single query, for which the client should receive
+/// receive relevant `TransactionUpdate`s.
+///
+/// After issuing a `SubscribeSingle` message, the client will receive a single
+/// `SubscribeApplied` message containing every current row which matches the query. Then, any
+/// time a reducer updates the query's results, the client will receive a `TransactionUpdate`
+/// containing the relevant updates.
+///
+/// If a client subscribes to queries with overlapping results, the client will receive
+/// multiple copies of rows that appear in multiple queries.
+#[derive(SpacetimeType)]
+#[sats(crate = spacetimedb_lib)]
+pub struct SubscribeSingle {
+    /// A single SQL `SELECT` query to subscribe to.
+    pub query: Box<str>,
+    /// An identifier for a client request.
+    /// This should not be reused for any other subscriptions on the same connection.
+    /// TODO: Can we call this subscription_id? It feels odd that this request_id will be used for multiple messages (the Unsubscribe call).
+    pub request_id: u32,
+}
+
+/// Client request for removing a query from a subscription.
+#[derive(SpacetimeType)]
+#[sats(crate = spacetimedb_lib)]
+pub struct Unsubscribe {
+    /// An identifier for a client request.
+    pub request_id: u32,
+    /// The ID returned in the [`SubscribeApplied`] message.
+    /// An optimization to avoid reparsing and normalizing the query string.
+    /// TODO: I assume this is just used for debugging?
+    pub query_id: QueryId,
```

Comment:
See previous comment. This `query_id` is (supposed to be) the thing you want to call `subscription_id`.